### PR TITLE
MGMT-5244: Only include 4.8+ in operator manifests

### PR DESCRIPTION
**Pull request by djzager**
Since the assisted-service, when deployed via the operator, only
supports OpenShift versions >= 4.8 we will only include 4.8+ in the
operator manifests.

Update the `hack/generate.sh` script to:
1) Delete versions 4.6 and 4.7 from the OPENSHIFT_VERSIONS json string
   for operator manifests.
2) Add inline python to update the agentserviceconfig sample using the
   evaluated openshift verions.

### CODE REVIEW

**Comment by djzager** [comment]
cc @mhrivnak @omertuc

File: hack/generate.sh
Comment (omertuc) [inline hack/generate.sh:83]: You can let it span multiple lines like in [here](https://github.com/openshift/release/blob/3f722a154be48991c3f65299b7c7249cb2a09b9a/ci-operator/step-registry/single-node/conf/aws/single-node-conf-aws-commands.sh#L12)

**Comment by omertuc** [comment]
Looks fine, not familiar enough with the alm-examples part so I'll wait for someone else to lgtm

File: hack/generate.sh
Comment (flaper87) [inline hack/generate.sh:83]: Would using `jq` to generate the desired json and then `sed` work? 

This is a snippet with a `jq` filter that should the same thing this python script does:

```
echo '{"4.8":{"display_name":"4.8.0-fc.3","release_version":"4.8.0-fc.3","release_image":"quay.io/openshift-release-dev/ocp-release:4.8.0-fc.3-x86_64","rhcos_image":"https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/pre-release/4.8.0-fc.3/rhcos-4.8.0-fc.3-x86_64-live.x86_64.iso","rhcos_rootfs":"https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/pre-release/4.8.0-fc.3/rhcos-live-rootfs.x86_64.img","rhcos_version":"48.84.202105062123-0","support_level":"beta"}}' | jq '[. | to_entries | .[] | {"openshiftVersion": .key, "version": .value["rhcos_version"], "url": .value["rhcos_image"], "rootFSUrl": .value["rhcos_rootfs"] }]'
[
  {
    "openshiftVersion": "4.8",
    "version": "48.84.202105062123-0",
    "url": "https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/pre-release/4.8.0-fc.3/rhcos-4.8.0-fc.3-x86_64-live.x86_64.iso",
    "rootFSUrl": "https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/pre-release/4.8.0-fc.3/rhcos-live-rootfs.x86_64.img"
  }
]
```

You can play with it here: https://jqplay.org/s/ABfiwfFacJ

**Review by flaper87 — CHANGES_REQUESTED** [review]

File: hack/generate.sh
Comment (djzager) [reply hack/generate.sh:83]: is there a way to transform the json to yaml ?

File: config/manager/manager.yaml
Comment (slaviered) [inline config/manager/manager.yaml:44]: It should also have default:true, just for good measures (although its not going to reach the UI)

File: config/samples/agent-install.openshift.io_v1beta1_agentserviceconfig.yaml
Comment (slaviered) [inline config/samples/agent-install.openshift.io_v1beta1_agentserviceconfig.yaml:4]: Why changing this?

File: deploy/olm-catalog/manifests/assisted-service-operator.clusterserviceversion.yaml
Comment (slaviered) [inline deploy/olm-catalog/manifests/assisted-service-operator.clusterserviceversion.yaml:439]: Same as above with the default field

File: config/manager/manager.yaml
Comment (djzager) [reply config/manager/manager.yaml:44]: We would need to make it the default in https://github.com/openshift/assisted-service/blob/master/data/default_ocp_versions.json 
This only takes that stuff and trims it to 4.8

File: config/samples/agent-install.openshift.io_v1beta1_agentserviceconfig.yaml
Comment (djzager) [reply config/samples/agent-install.openshift.io_v1beta1_agentserviceconfig.yaml:4]: I believe the original was spaced incorrectly. Whatever the reason, It got updated by the little `python -c` and I don't think it's worth fighting.

File: deploy/olm-catalog/manifests/assisted-service-operator.clusterserviceversion.yaml
Comment (djzager) [reply deploy/olm-catalog/manifests/assisted-service-operator.clusterserviceversion.yaml:439]: Same as above, this would need to be set in https://github.com/openshift/assisted-service/blob/master/data/default_ocp_versions.json

File: hack/generate.sh
Comment (omertuc) [reply hack/generate.sh:83]: Technically JSON is valid YAML

Comment (flaper87) [reply hack/generate.sh:83]: > is there a way to transform the json to yaml ?

Plenty, but let's not hold this PR on that. 

Personally, I think this snippet is not very maintainable, or readable. I would like us to use the tools available in the dev environments/systems we have.

> Technically JSON is valid YAML

@omertuc I am sure @djzager knows this :smile: the question was related to having the example be fully `YAML` as his script is currently generating it. In your defense, @djzager's comment was very succint. 

Let's try to provide more context in these comments to help conversations move forward

**Review by flaper87 — APPROVED** [review]
Let's not hold the PR on my comment but, I would really appreciate it if @djzager could switch from the script used in this PR to using some of the CLI tools available already in a follow-up PR.

**Pull request merged**
